File: Snooper/src/main/java/com/prateekj/snooper/dbreader/adapter/TableAdapter.java
package com.prateekj.snooper.dbreader.adapter;

import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.prateekj.snooper.R;

import java.util.List;

public class TableAdapter extends RecyclerView.Adapter<TableAdapter.TableViewHolder> {

  private List<String> tableList;
  private TableEventListener tableEventListener;

  class TableViewHolder extends RecyclerView.ViewHolder {
    TextView name;
    TextView rowNum;

    TableViewHolder(View view) {
      super(view);
      name = (TextView) view.findViewById(R.id.table_name);
      rowNum = (TextView) view.findViewById(R.id.row_num);
    }

    void bind(final String tableName, int rowNum) {
      this.name.setText(tableName);
      this.rowNum.setText(Integer.toString(rowNum) + ". ");
      this.itemView.setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View v) {
          tableEventListener.onTableClick(tableName);
        }
      });
    }
  }

  public TableAdapter(List<String> tableList, TableEventListener tableEventListener) {
    this.tableList = tableList;
    this.tableEventListener = tableEventListener;
  }

  @Override
  public TableViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    View itemView = LayoutInflater.from(parent.getContext())
      .inflate(R.layout.table_item, parent, false);
    return new TableViewHolder(itemView);
  }

  @Override
  public void onBindViewHolder(final TableViewHolder holder, int position) {
    holder.bind(tableList.get(position), position + 1);
  }

  @Override
  public int getItemCount() {
    return tableList.size();
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/dbreader/adapter/TableAdapter.kt
package com.prateekj.snooper.dbreader.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.prateekj.snooper.R
import kotlinx.android.synthetic.main.table_item.view.*

class TableAdapter(
  private val tableList: List<String>,
  private val tableEventListener: TableEventListener
) : RecyclerView.Adapter<TableAdapter.TableViewHolder>() {

  class TableViewHolder(
    val view: View,
    private val tableEventListener: TableEventListener
  ) : RecyclerView.ViewHolder(view) {

    fun bind(tableName: String, rowNum: Int) {
      view.table_name.text = tableName
      view.row_num.text = "$rowNum. "
      this.itemView.setOnClickListener { tableEventListener.onTableClick(tableName) }
    }
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TableViewHolder {
    val itemView = LayoutInflater.from(parent.context)
      .inflate(R.layout.table_item, parent, false)
    return TableViewHolder(itemView, tableEventListener)
  }

  override fun onBindViewHolder(holder: TableViewHolder, position: Int) {
    holder.bind(tableList[position], position + 1)
  }

  override fun getItemCount(): Int {
    return tableList.size
  }
}